api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -35,4 +35,20 @@ message UdpProxyConfig {
   // The idle timeout for sessions. Idle is defined as no datagrams between received or sent by
   // the session. The default if not specified is 1 minute.
   google.protobuf.Duration idle_timeout = 3;
+
+  // Use the remote downstream ip address as sender ip address when send packets to upstreams.
```

Comment:
> I did some light editing.
> 
> ```
> // Use the remote downstream IP address as the sender IP address when sending packets to upstream hosts.
> // This option requires Envoy to be run with the *CAP_NET_ADMIN* capability on Linux.
> // This option does not preserve the remote downstream port.
> // If this option is enabled, the IP address of sent datagrams will be changed to the remote downstream IP address.
> // This means that Envoy will not receive packets that are sent by upstream hosts because the upstream hosts
> // will send the packets with the remote downstream IP address as the destination. All packets will be routed
> // to the remote downstream directly if there are route rules on the upstream host side.
> // There are two options to return the packets back to the remote downstream.
> // The first one is to use DSR (Direct Server Return).
> // The other one is to configure routing rules on the upstream hosts to forward
> // all packets back to Envoy and configure iptables rules on the host running Envoy to
> // forward all packets from upstream hosts to the Envoy process so that Envoy can forward the packets to the downstream.
> // If there are errors while setting this option on sessions, Envoy will fall back to not using `use_original_src_ip` mode even if this option is set.
> // If the platform does not support this option, Envoy will raise a configuration error.
> ```

Thanks your fixes. :)

> Out of curiosity, without the port being transparently forwarded this option seems of limited utility? What do you plan on using it for? Is there a way we can forward port in the future potentially via raw socket mode on Linux?

By my understanding, there is no way to modify source port number because Linux kernel does not support it on UDP stack.
But I think that we can forward port as well if we send packets via raw socket mode on Linux.

I'm working on 5G telecom for core networks but there is no requirement for forwarding port as well yet.
If there are some requirement for forwarding port in the feature on telecom world, I can add that feature for that maybe. :)